include/aspect/lateral_averaging.h

```diff
@@ -0,0 +1,195 @@
+/*
+  Copyright (C) 2011-2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__lateral_averaging_h
+#define __aspect__lateral_averaging_h
+
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  /**
+   * LateralAveraging is a class that performs various averaging operations
+   * on the solution.  The functions of this class take a vector as an argument.
+   * The model is divided into depth slices where the number of slices is the
+   * length of the input vector, and then the relevant quantity is averaged
+   * laterally for each depth slice.
```

Comment:
What you call the "input vector" is actually the output vector, I suppose, because you write into it. What you term "relevant quantity" is given by the name of the function?